independent-projects/bootstrap/runner/src/main/java/io/quarkus/bootstrap/runner/JarFileReference.java

```diff
@@ -90,11 +139,13 @@ static <T> T withJarFile(JarResource jarResource, String resource, JarFileConsum
 
         // Happy path: the jar reference already exists and it's ready to be used
         final var localJarFileRefFuture = jarResource.jarFileReference.get();
+        boolean closingLocalJarFileRef = false;
         if (localJarFileRefFuture != null && localJarFileRefFuture.isDone()) {
             JarFileReference jarFileReference = localJarFileRefFuture.join();
             if (jarFileReference.acquire()) {
                 return consumeSharedJarFile(jarFileReference, jarResource, resource, fileConsumer);
             }
+            closingLocalJarFileRef = true;
```

Comment:
The acquire failed implies that the reference is already marked to be closed.

Reply:
Again